test/topology_custom/test_mv_large_inserts.py

```diff
@@ -0,0 +1,61 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+from test.pylib.manager_client import ManagerClient
+
+import asyncio
+import pytest
+import time
+import logging
+from test.topology.conftest import skip_mode
+from cassandra.cqltypes import Int32Type
+
+logger = logging.getLogger(__name__)
+
+async def insert_with_concurrency(cql, concurrency):
+    def serialize_int(i):
+        return Int32Type.serialize(i, cql.cluster.protocol_version)
+    def get_replicas(key):
+        return cql.cluster.metadata.get_replicas("ks", serialize_int(key))
+    for key in range(100):
+        logger.info(f"Starting writes with concurrency {concurrency}")
+        async def do_inserts(i: int):
+            key = i
+            local_node = get_replicas(key)[0]
+            while local_node == get_replicas(i)[0]:
+                i = i + concurrency
+            await cql.run_async(f"INSERT INTO ks.tab (key, c, v) VALUES ({key}, {i}, '{10000 * "a"}')")
+
+        tasks = [asyncio.create_task(do_inserts(i)) for i in range(concurrency)]
+        await asyncio.gather(*tasks)
+        logger.info(f"Finished writes with concurrency {concurrency}")
+
+async def wait_for_views(cql, mvs_count, node_count):
+    deadline = time.time() + 120
+    while time.time() < deadline:
+        done = await cql.run_async(f"SELECT COUNT(*) FROM system_distributed.view_build_status WHERE status = 'SUCCESS' ALLOW FILTERING")
+        logger.info(f"Views built: {done[0][0]}")
+        if done[0][0] == node_count * mvs_count:
+            return
+        else:
+            time.sleep(0.2)
+    raise Exception("Timeout waiting for views to build")
+
+
+@pytest.mark.asyncio
+@skip_mode('release', "error injections aren't enabled in release mode")
+async def test_large_concurrency(manager: ManagerClient) -> None:
+    node_count = 2
+    await manager.servers_add(node_count, cmdline=['--error-injections-at-startup=view_update_limit'])
+    cql = manager.get_cql()
+    await cql.run_async(f"CREATE KEYSPACE ks WITH replication = {{'class': 'SimpleStrategy', 'replication_factor': 1}}")
+    await cql.run_async(f"CREATE TABLE ks.tab (key int, c int, v text, PRIMARY KEY (key, c))")
+    await cql.run_async(f"CREATE MATERIALIZED VIEW ks.mv_cf_view AS SELECT * FROM ks.tab "
+                    "WHERE c IS NOT NULL and key IS NOT NULL PRIMARY KEY (c, key) ")
+    await wait_for_views(cql, 1, node_count)
+    with pytest.raises(Exception, match="View update backlog is too high"):
+        await insert_with_concurrency(cql, 1000)
```

Comment:
Interesting, the CQL driver doesn't retry a query automatically if it fails because of overload?

Reply:
I looks like it does, so I'm not sure how this test was passing, but it wasn't consistent for me so I'm changing the retry policy in the new test anyway